exporters/tests/test_pelorus_base_plugin.py

```diff
@@ -0,0 +1,342 @@
+    assert http_exception.typename == "HTTPException"
+
+    assert http_exception.value.detail == "pong"
+    assert http_exception.value.status_code == http.HTTPStatus.OK
+
+
+def test_abstract_classes():
+    """
+    Test for the plugin that did not implement required abstract methods.
+    """
+
+    class MyPelorusWebhookPlugin(PelorusWebhookPlugin):
+        def __init__(self, handshake_headers: Headers, request: Request) -> None:
+            super().__init__(handshake_headers, request)
+
+    # We should get an error:
+    # TypeError: Can't instantiate abstract class MyPelorusWebhookPlugin with abstract
+    #            methods _receive_pelorus_payload, handshake
+    with pytest.raises(TypeError) as type_error:
+        MyPelorusWebhookPlugin(None, None)
+
+    assert (
+        str(type_error.value)
+        == "Can't instantiate abstract class MyPelorusWebhookPlugin "
+        + "with abstract methods _handshake, _receive_pelorus_payload"
+    )
+
+
+class SimplePelorusWebhookPlugin(PelorusWebhookPlugin):
+    @override
+    async def _handshake(self) -> Awaitable[bool]:
+        pass
+
+    @override
+    async def _receive_pelorus_payload(self, Any) -> Awaitable[PelorusMetric]:
+        pass
+
+
+@pytest.mark.parametrize(
+    "handshake_headers,request_data",
+    [
+        ("Header.", "request data"),
+    ],
+)
+def test_pelorus_webhook_plugin_abc(handshake_headers, request_data):
+    """
+    Test for the plugin that did implement required abstract methods.
+    """
+
+    plugin_instance = SimplePelorusWebhookPlugin(
+        handshake_headers=handshake_headers, request=request_data
+    )
+
+    assert plugin_instance.headers == handshake_headers
+    assert plugin_instance.request == request_data
+
+
+def test_check_is_pelorus_webhook_handler():
+    """
+    Test if the plugin is a proper Pelorus handler.
+    """
+
+    class MyPelorusWebhookPlugin(PelorusWebhookPlugin):
+        user_agent_str = "Pelorus-Webhook/"
+
+    assert MyPelorusWebhookPlugin.is_pelorus_webhook_handler()
+
+
+def test_check_is_pelorus_not_webhook_handler():
+    """
+    Test if the plugin is not a proper Pelorus handler.
+    """
+
+    class MyPelorusWebhookPlugin(PelorusWebhookPlugin):
+        pass
+
+    assert not MyPelorusWebhookPlugin.is_pelorus_webhook_handler()
+
+
+class WithUserAgentWebhookPlugin(PelorusWebhookPlugin):
+    user_agent_str = "Pelorus-Webhook/"
+
+
+class WithoutUserAgentWebhookPlugin(PelorusWebhookPlugin):
+    pass
+
+
+class EmptyUserAgentWebhookPlugin(PelorusWebhookPlugin):
+    user_agent_str = ""
+
+
+def test_check_can_handle_methods():
+    """
+    Tests if the plugin can handle data based on the provided
+    string information that normally is given by the
+    "User-Agent:" header information.
+
+    Pelorus Webhook loads many plugins so each plugin should
+    know what payload data can it handle.
+    """
+
+    assert WithUserAgentWebhookPlugin.can_handle("Pelorus-Webhook/")
```

Comment:
I think it is hard to read like this, I would have a test for each class

Reply:
created line between With and Without and added comment:

```python

    # user_agent_str = "Pelorus-Webhook/"
    assert WithUserAgentWebhookPlugin.can_handle("Pelorus-Webhook/")
    assert WithUserAgentWebhookPlugin.can_handle("Pelorus-Webhook/suffix")
    assert not WithUserAgentWebhookPlugin.can_handle("")
    assert not WithUserAgentWebhookPlugin.can_handle(None)
    assert not WithUserAgentWebhookPlugin.can_handle("Incompatible-Webhook")

    # user_agent_str not defined
    assert not WithoutUserAgentWebhookPlugin.can_handle(None)
    assert not WithoutUserAgentWebhookPlugin.can_handle("")
```